Failing because of this..need to use npm i think!

```
npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to https://artifactory.global.square/artifactory/api/npm/square-npm/fastparse/-/fastparse-1.1.2.tgz failed, reason: getaddrinfo ENOTFOUND artifactory.global.square
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
npm ERR! A complete log of this run can be found in: /vercel/.npm/_logs/2024-06-26T18_51_52_157Z-debug-0.log
Error: Command "npm install" exited with 1
```

--- a/package.json
+++ b/package.json
@@ -20,6 +20,7 @@
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.1",
     "crypto-browserify": "^3.12.0",
+    "daisyui": "^4.12.8",
     "lucide-react": "^0.396.0",
     "moment": "^2.30.1",
     "next": "14.2.4",

--- a/src/components/calendarpage.jsx
+++ b/src/components/calendarpage.jsx
@@ -1,51 +1,50 @@
 import FullCalendar from '@fullcalendar/react';
 import dayGridPlugin from '@fullcalendar/daygrid'
 import { useState } from "react";
+import Period from "@/components/period"
+import { PeriodDialog } from "@/components/period-dialog"
 
 // TODO: Read events from DWN
 const startingEvents = [
-  { title: 'event 1', date: '2024-06-23' },
-  { title: 'event 2', date: '2024-06-24' }
+  { title: 'event 1', date: '2024-06-12' },
+  { title: 'event 2', date: '2024-06-14' }
 ]
+
+const formatDateForCalendar = (date) => date.toISOString().split('T')[0]
 
 export function CalendarPage() {
   const [events, setEvents] = useState(startingEvents);
-
-  // TODO: Delete me later. For demo purposes
-  function getRandomJuneDate() {
-    const year = 2024;
-    const month = 6; // June
-    const minDay = 1;
-    const maxDay = 30;
-
-    // Generate a random day in June
-    const day = Math.floor(Math.random() * (maxDay - minDay + 1)) + minDay;
-
-    // Format the date as "YYYY-MM-DD"
-    const formattedDay = day.toString().padStart(2, '0'); // Ensure two digits for the day
-    const formattedDate = `${year}-${month.toString().padStart(2, '0')}-${formattedDay}`;
-
-    return formattedDate;
-  }
+  const [periodStartDate, setPeriodStartDate] = useState(null);
+  const [dialogTitle, setDialogTitle] = useState('Dialog');
+  const [isOpen, setIsOpen] = useState(false);
+  const [dialogEvent, setDialogEvent] = useState(null); // string
 
   // TODO: Replace with data from the modal submission
-  function addEvent() {
-    console.log("Adding an event");
+  function addEvent(title, date) {
+    console.log("Adding a specific event");
+    console.log("events: ", events, title, date);
+    setEvents([
+      ...events,
+      {
+        date: date,
+        title: title,
+      }
+    ]);
+    console.log("events: ", events);
+  }
+
+  function addRandomEvent(title) {
+    console.log("Adding a random event");
+    console.log("events: ", events);
 
     setEvents([
       ...events,
       {
-        date: getRandomJuneDate(),
-        title: "Random event",
+        date: formatDateForCalendar(new Date()),
+        title: title ,
       }
     ])
   }
-
-  const handleEventClick = (info) => {
-    console.log("Handling event click");
-    console.log(info);
-    // TODO: Display modal with event details
-  };
 
   const EventItem = ({ info }) => {
     // TODO: Customize how the event is displayed on the calendar
@@ -57,18 +56,72 @@ export function CalendarPage() {
     );
   };
 
+  const formatDate = (date) => {
+    if (!date) return '';
+    let d = new Date(date),
+        month = '' + (d.getMonth() + 1),
+        day = '' + d.getDate(),
+        year = d.getFullYear();
+
+    if (month.length < 2) 
+        month = '0' + month;
+    if (day.length < 2) 
+        day = '0' + day;
+
+    return [year, month, day].join('-');
+  };
+
+  function onDialogClose(localStartDate, localEndDate) {
+    console.log("onDialogClose");
+    setIsOpen(false);
+    // TODO: Update the event if the localStartDate changed
+    
+    // TODO: Add events between localStartDate and localEndDate
+    addEvent("end", localEndDate);
+  }
+
+  const startPeriod = () => {
+    const now = new Date();
+    setPeriodStartDate(now);
+    addRandomEvent(!periodStartDate ? "Period started" : "Period ended");
+  }
+
+  // const openModal = () => {document.getElementById('my_modal_1').showModal()}
+  console.log("rendering calendar page")
+
+  const updateModal = (newContent) => {
+    const event = newContent.event
+    const title = event.title
+    setDialogEvent(event);
+    setDialogTitle(title);
+    setIsOpen(true);
+  }
+
   return (
     <div className="flex flex-col min-h-[100dvh]">
+      <Period startDate={periodStartDate} setStartDate={startPeriod} />
       <FullCalendar
         selectable={true}
         plugins={[dayGridPlugin]}
-        eventClick={handleEventClick}
+        eventClick={updateModal}
         eventContent={(info) => <EventItem info={info} />}
         initialView="dayGridMonth"
         events={events}
       />
+      {/* <button className="btn" onClick={openModal}>open modal</button> */}
+      {/* <Modal> */}
 
-      <button onClick={addEvent}>Add random event</button>
+        {/* <div className="modal-box">
+        <h3 className="font-bold text-lg">Hello Tal!</h3>
+        <p className="py-4">Press ESC key or click the button below to close</p>
+        <div className="modal-action">
+          <form method="dialog">
+            <button className="btn">Close</button>
+          </form>
+        </div>
+      </div> */}
+      {/* </Modal> */}
+      {isOpen && <PeriodDialog title={dialogTitle} event={dialogEvent} onClose={onDialogClose} />}
     </div>
   )
 };

--- a/src/components/modal.jsx
+++ b/src/components/modal.jsx
@@ -1,0 +1,35 @@
+
+import React, { useState } from 'react';
+
+
+export function Modal({children}) {
+  return (
+    <dialog id="my_modal_1" className="modal">
+      {children}
+    </dialog>
+  )
+}
+
+
+export const DynamicDialog = ({title, dialogContent, onClose}) => {
+
+  return (
+    <div className="p-4">
+        <div className="modal modal-open">
+          <div className="modal-box">
+            <h3 className="font-bold text-lg">{title}</h3>
+            <p className="py-4">{dialogContent}</p>
+            <div className="modal-action">
+              <button
+                className="btn"
+                onClick={onClose}
+              >
+                Close
+              </button>
+            </div>
+          </div>
+        </div>
+    </div>
+  );
+};
+

--- a/src/components/period-dialog.jsx
+++ b/src/components/period-dialog.jsx
@@ -1,0 +1,45 @@
+import { useState } from "react";
+import { DynamicDialog } from "@/components/modal"
+
+const formatDate = (date) => {
+  if (!date) return '';
+  let d = new Date(date),
+      month = '' + (d.getMonth() + 1),
+      day = '' + d.getDate(),
+      year = d.getFullYear();
+
+  if (month.length < 2) 
+      month = '0' + month;
+  if (day.length < 2) 
+      day = '0' + day;
+
+  return [year, month, day].join('-');
+};
+
+export const PeriodDialog = ({ dialogTitle, event, onClose }) => {
+  const [localStartDate, setLocalStartDate] = useState(formatDate(event.start));
+  const [localEndDate, setLocalEndDate] = useState(formatDate(event.end));
+
+  const handleStartDateChange = (e) => {
+    setLocalStartDate(e.target.value);
+  };
+  const handleEndDateChange = (e) => {
+    setLocalEndDate(e.target.value);
+  };
+
+  const handleClose = () => {
+    onClose(localStartDate, localEndDate);
+  };
+
+  const dialogContent = (
+      <div>
+        <label htmlFor="start">Started: </label>
+        <input type="date" id="start" name="start" value={localStartDate} onChange={handleStartDateChange} />
+        <br/>
+        <label htmlFor="end">Ended: </label>
+        <input type="date" id="end" name="end" value={localEndDate} onChange={handleEndDateChange} />
+      </div>
+    );
+  
+  return (<DynamicDialog title={dialogTitle} dialogContent={dialogContent} onClose={handleClose} />);
+}

--- a/src/components/period.jsx
+++ b/src/components/period.jsx
@@ -1,0 +1,17 @@
+import React from "react";
+import moment from "moment"
+
+export default function Period({ startDate, setStartDate }) {
+  return <div>
+
+    {!startDate && (
+      <button className="btn btn-primary" onClick={() => setStartDate(moment())}>My period started</button>
+    )}
+    {startDate && (
+      <div>
+        <span>My current period started on: {startDate.toString()}</span><br/>
+        <button className="btn btn-primary" onClick={() => setStartDate(null)}>My period ended</button>
+      </div>
+    )}
+  </div>
+}

--- a/tailwind.config.js
+++ b/tailwind.config.js
@@ -20,8 +20,8 @@ module.exports = {
             "background": "hsl(var(--background))",
             "foreground": "hsl(var(--foreground))",
             "primary": {
-              "DEFAULT": "hsl(var(--primary))",
-              "foreground": "hsl(var(--primary-foreground))"
+              "DEFAULT": "var(--primary)",
+              "foreground": "var(--primary-foreground)"
             },
             "secondary": {
               "DEFAULT": "hsl(var(--secondary))",
@@ -72,5 +72,5 @@ module.exports = {
             }
           }
 },
-  plugins: [require("tailwindcss-animate")],
+  plugins: [require("tailwindcss-animate"), require("daisyui")],
 };

